### PR TITLE
Dev data procedure doc

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: docs/data_pipeline_procedure.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          --source_ver $source_verion \
+          --token_filepath $token_filepath \
+          --retrieve_sample 
+        ```
+     * Generated files/metafiles
+       * downloaded files
+         ```text
+         sourcedata/WS/[env]/[workspace_id]/[UPA]/
+         ```
+       * Metadata files
```
Comment (MrCreosote) [inline docs/data_pipeline_procedure.md:20]:
```suggestion
       * Metadata file
```

File: docs/data_pipeline_procedure.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     ```
+   * Generated files/metafiles
+     * Taskfarmer job files
+       ```text
+       task_farmer_jobs/[job_dir]/
+       ```
+     * Taskfarmer jobs tracking file
+       ```text
+       task_farmer_jobs/task_info.jsonl
+       ```
```
Comment (MrCreosote) [inline docs/data_pipeline_procedure.md:69]: genome IDs file?

File: docs/data_pipeline_procedure.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+       --kbase_collection $kbase_collection \
+       --source_ver $source_ver \
+       --load_ver $load_ver \
+       --source_file_ext $source_file_ext \
+       --submit_job
+     ```
+   * Generated files/metafiles
+     * Taskfarmer job files
+       ```text
+       task_farmer_jobs/[job_dir]/
```
Comment (MrCreosote) [inline docs/data_pipeline_procedure.md:64]: Can you explain the value of `job_dir` and what's contained in the dir?

File: docs/data_pipeline_procedure.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+       ```
```
Comment (MrCreosote) [inline docs/data_pipeline_procedure.md:65]: Is there not an sdk job dir as well?

File: docs/data_pipeline_procedure.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+       collectionsdata/[env]/[kbase_collection]/[load_ver]/[tool]/[batch_dir]/genome_metadata.tsv
+       ```
+## Parse Tool Results
+   * Example usage
+     ```commandline
+     python src/loaders/genome_collection/parse_tool_results.py \
+       --env $env \
+       --kbase_collection $kbase_collection \
+       --source_ver $source_ver \
+       --load_ver $load_ver
```
Comment (MrCreosote) [inline docs/data_pipeline_procedure.md:86]: Strictly speaking if you know the load ver you know the source ver as well, or at least you should. Not sure how easy that is to figure out based on the files we have now, but in the DB it should be an explicit relation to my mind

**Review by MrCreosote — COMMENTED** [review]
I have a few questions but otherwise looks good

@Xiangs18 should take a look as well

File: docs/data_pipeline_procedure.md (hunk 1 above)
Comment (Tianhao-Gu) [reply docs/data_pipeline_procedure.md]: 👍

File: docs/data_pipeline_procedure.md (hunk 2 above)
Comment (Tianhao-Gu) [reply docs/data_pipeline_procedure.md:80]: added

File: docs/data_pipeline_procedure.md (hunk 5 above)
Comment (Tianhao-Gu) [reply docs/data_pipeline_procedure.md:97]: yea. I don't remember exactly why we keep 'source_ver' for parser. I think we can eliminate 'source_ver' from the parser script. It seems like it's only used to get the sample files which are also softlinked to collectionssource directory.

Comment (MrCreosote) [reply docs/data_pipeline_procedure.md:97]: Also to get mash files I think

File: docs/data_pipeline_procedure.md (hunk 1 above)
Comment (MrCreosote) [reply docs/data_pipeline_procedure.md]: :+1:

File: docs/data_pipeline_procedure.md (hunk 3 above)
Comment (MrCreosote) [reply docs/data_pipeline_procedure.md:69]: What is the actual value of `job_dir` though? Is it a job id? If so where does it come from?

File: docs/data_pipeline_procedure.md (hunk 4 above)
Comment (MrCreosote) [reply docs/data_pipeline_procedure.md:70]: :+1:

File: docs/data_pipeline_procedure.md (hunk 2 above)
Comment (MrCreosote) [reply docs/data_pipeline_procedure.md:80]: Are those in the taskfarmer job dir? I thought there was one file per tool instance run?

File: docs/data_pipeline_procedure.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+   * Generated files/metafiles
+     * Taskfarmer job files
+       ```text
+       task_farmer_jobs/[job_dir]/
+       ```
+       * including files utilized by TaskFarmer
+         * shifter_wrapper.sh
+         * submit_taskfarmer.sl
+         * tasks.txt
+         * genome_id_x.tsv
```
Comment (MrCreosote) [inline docs/data_pipeline_procedure.md:75]: Can you provide a very brief description of these 2 files?

File: docs/data_pipeline_procedure.md (hunk 3 above)
Comment (Tianhao-Gu) [reply docs/data_pipeline_procedure.md:69]: added.

File: docs/data_pipeline_procedure.md (hunk 6 above)
Comment (Tianhao-Gu) [reply docs/data_pipeline_procedure.md]: 👍

File: docs/data_pipeline_procedure.md (hunk 2 above)
Comment (MrCreosote) [reply docs/data_pipeline_procedure.md:81]: :+1:

File: docs/data_pipeline_procedure.md (hunk 3 above)
Comment (MrCreosote) [reply docs/data_pipeline_procedure.md:69]: :+1:

**Review by MrCreosote — DISMISSED** [review]
If @Xiangs18 approves so do I

**Comment by Xiangs18** [comment]
It seems we are missing compute_genome_taxa_count.py. Do we need to include it as well?

**Comment by Tianhao-Gu** [comment]
> It seems we are missing compute_genome_taxa_count.py. Do we need to include it as well?

@Xiangs18 @MrCreosote added compute taxa count section.

File: docs/data_pipeline_procedure.md [hunk 7]
```diff
@@ -89,16 +89,33 @@
...
+     * Generated files/metafiles
+       * Parsed JSONL files for ArangoDB import
+         ```text
+         import_files/[env]/
+         ```
+   * Parse genome taxa count
+     * Example usage
+       ```commandline
+       python src/loaders/genome_collection/compute_genome_taxa_count.py \
+         ../import_files/$env/GROW_tian_test_taskfarmer_checkm2_gtdb_tk_kbcoll_genome_attribs.jsonl \
```
Comment (Xiangs18) [inline docs/data_pipeline_procedure.md:110]: Maybe `GROW_2023.06_checkm2_gtdb_tk_computed_genome_attribs.jsonl`?

Comment (Tianhao-Gu) [reply docs/data_pipeline_procedure.md]: 👍

**Review by MrCreosote — APPROVED** [review]

**Review by Xiangs18 — APPROVED** [review]

**Pull request merged**